I3uckwheat/cards-of-carousal
# Feature/139 get packs from api

#### 0. Make sure your issue is linked:

Closes #139 . Closes #153 .

#### 1. Purpose:


Fetches card packs selected in the host game settings menu when Start Carousing button is clicked.

#### 2. Implementation:


Once "Start Carousing" is click (must have at least one player in lobby and at least one card pack selected), the dispatch will send the selected packs array to the async reducer, which fetches the cards from our card API endpoint. It then places those cards in the host context under the property `deck`. This property is an object consisting of a `black` and `white` array.

#### 3. Possible Issues:


none

#### 4. How to Test:


1. Pull this branch.
2. Run `npm dev:frontend` and `npm dev:backend` in separate terminals.
3. Open a second browser window with `http://localhost:3000` in the address bar.
4. In the first window, click 'Host'. Get the lobby ID. 
5. In the second window, click 'Join', type in a name and the lobby ID, and join.
6. In the first window, select any card packs you want.
7. Click "Start Carousing", open chrome dev tools, inspect the Context.Provider > value > state > deck to ensure cards were delivered.


#### 5. Cleanup
